src/librustc/traits/select.rs

```diff
@@ -2081,6 +2125,11 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 substs.upvar_tys(def_id, self.tcx()).collect()
             }
 
+            ty::TyGenerator(def_id, ref substs, interior) => {
+                let witness = iter::once(interior.witness());
```

Comment:
You can leave the obligation ambiguous if the witness isn't resolved yet (by setting `ambiguous = true` from `assemble_candidates_from_default_impls`).